android/app/src/main/java/com/mattermost/rnbeta/RNPasteableEditTextOnPasteListener.java

```diff
@@ -119,4 +133,31 @@ public void onPaste(Uri itemUri) {
                         event
                 );
     }
+
+    private String moveToImagesCache(String src, String fileName) {
+        ReactContext ctx = (ReactContext)mEditText.getContext();
+        String dest = ctx.getCacheDir().getAbsolutePath() + "/Images/" + fileName;
+        File srcFile = new File(src);
+        InputStream in = null;
+        OutputStream out = null;
+
+        try {
+            in = new FileInputStream(srcFile);
+            out = new FileOutputStream(dest);
+
+            byte[] buf = new byte[10240];
```

Comment:
Using a `BufferedInputStream` instead (or around?) a `FileInputStream` could automatically take care of buffering the file being read without having to resort to a manually-maintained value (`10240`) for buffer size per read piece. Android's default is 8kb (I think) which should be plenty.

Reply:
https://stackoverflow.com/questions/18600331/why-is-using-bufferedinputstream-to-read-a-file-byte-by-byte-faster-than-using-f/18600383

Reply:
Or better yet, `Files.copy` since Java 7 and above.

`Files.copy(source.toPath(), dest.toPath());` (throws `IOException`)

Performant, and saves the trouble of manually managing stream closing and flushing (which I often forget).

Reply:
There's `Files.move` too which may remove the need to manually delete the source file.